### PR TITLE
Improve security and UX

### PR DESCRIPTION
**Pull request by erkanerol**
Towards https://github.com/giantswarm/roadmap/issues/2090

On top of #30 

This PR is the third step of the road map described in the issue. I am able to deploy CPI and CSI by providing only these values.
```yaml
global:
  config:
    clusterId: <my-cluster-name>
    vcenter: "<our-vcenter>.ionoscloud.tools"
    username: "vcenter-admin@vsphere.local"
    password: "<my-password>"
    datacenter: "Datacenter"
    region: "ionos-region"
    zone: "ionos-zone"
    thumbprint: "F7:CF:F9:E5:99:39:FF:C1:D7:14:F1:3F:8A:42:21:95:3B:A1:6E:16"
```

This PR
- reorganizes values.yaml of subcharts. only keeps the common+credential related ones in the parent's one.
- generates the schema based on the new values.yaml
- updates CPI to 1.26.  See https://github.com/kubernetes/cloud-provider-vsphere/tree/release-1.26
- improves security for CPI by moving credentials from configmap to secret (I will open a PR in the upstream side too) 

Automation of manifest generation will be implemented with a separate PR.

### Testing

- [x] fresh install works
- [ ] upgrade from previous version works

### Checklist

- [x] Update changelog in CHANGELOG.md.
- [x] Make sure `values.yaml` and `values.schema.json` are valid.

### CODE REVIEW

File: helm/cloud-provider-vsphere/charts/cloud-provider-for-vsphere/values.yaml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+global:
+  config:
+    enabled: false
+
+config:
+  enabled: true
+  name: vsphere-cloud-config
+  vcenter: "vcenter.local"
+  username: "user"
+  password: "pass"
```
Comment (vxav) [inline helm/cloud-provider-vsphere/charts/cloud-provider-for-vsphere/values.yaml:14]: Should we remove user/password to enforce use of a secret like we do in other onprem providers?

File: helm/cloud-provider-vsphere/charts/cloud-provider-for-vsphere/templates/secret_config.yaml [hunk 2]
```diff
@@ -1,23 +1,26 @@
...
+stringData:
   vsphere.conf: |
     # Global properties in this section will be used for all specified vCenters unless overriden in VirtualCenter section.
     global:
       port: 443
+      {{- if .Values.config.tlsThumbprint }}
+      tlsThumbprint: {{- .Values.config.tlsThumbprint }}
+      {{- else }}
       # set insecure-flag to true if the vCenter uses a self-signed cert
       insecureFlag: true
```
Comment (vxav) [inline helm/cloud-provider-vsphere/charts/cloud-provider-for-vsphere/templates/secret_config.yaml:20]: We should not assume this value to always be true, most big customers will have valid certificates installed on their vCenters. Sounds better to make it configurable with `default: false`?

**Review by vxav — COMMENTED** [review]
Quite a lot going on. Couple minor things, lgtmiiwfy otherwise

File: helm/cloud-provider-vsphere/charts/cloud-provider-for-vsphere/values.yaml (hunk 1 above)
Comment (erkanerol) [reply helm/cloud-provider-vsphere/charts/cloud-provider-for-vsphere/values.yaml:14]: I plan to update this PR according to the discussion in slack regarding configuration interface. I forgot the discussion while opening this PR. Changing this PR as draft. 

Meanwhile you can give me an approval for this PR https://github.com/giantswarm/cloud-provider-vsphere-app/pull/30 :D

Comment (vxav) [reply helm/cloud-provider-vsphere/charts/cloud-provider-for-vsphere/values.yaml:14]: done

Comment (erkanerol) [reply helm/cloud-provider-vsphere/charts/cloud-provider-for-vsphere/values.yaml]: We decided to use this mechanism https://gigantic.slack.com/archives/C01F7T2MNRL/p1678703614073829

Also, we use this method for CAPVCD too. The only diff is we have only `refreshToken` there.

File: helm/cloud-provider-vsphere/charts/cloud-provider-for-vsphere/templates/secret_config.yaml (hunk 2 above)
Comment (erkanerol) [reply helm/cloud-provider-vsphere/charts/cloud-provider-for-vsphere/templates/secret_config.yaml]: Yeah. You are right.

I found this in the provider chart https://github.com/kubernetes/cloud-provider-vsphere/blob/master/charts/vsphere-cpi/templates/configmap.yaml#L16

And also I found there is a config in the cluster repo too https://github.com/giantswarm/cluster-vsphere/blob/main/helm/cluster-vsphere/values.yaml#L71

I added it to this repo: https://github.com/giantswarm/cloud-provider-vsphere-app/pull/31/files#diff-2c4c70a148aa47d79b8d3a24bae0247c998f22ecfe9e30e8ae59d4ee9c812ac4R12

Comment (vxav) [reply helm/cloud-provider-vsphere/charts/cloud-provider-for-vsphere/templates/secret_config.yaml]: Sorry I didn't specify, I meant `insecureFlag: true`
But it's also good to parameterise the thumbprint.

**Review by vxav — APPROVED** [review]

**Pull request merged**